charts/rancher-kiali-server/rancher-kiali-server/100.0.0+up1.35.0/templates/_helpers.tpl

```diff
@@ -50,8 +50,15 @@ app.kubernetes.io/part-of: "kiali"
 Selector labels
 */}}
 {{- define "kiali-server.selectorLabels" -}}
+{{- $releaseName := .Release.Name -}}
+{{- $fullName := include "kiali-server.fullname" . -}}
+{{- $deployment := (lookup "apps/v1" "Deployment" .Release.Namespace $fullName) -}}
 app.kubernetes.io/name: kiali
-app.kubernetes.io/instance: {{ include "kiali-server.fullname" . }}
+{{- if (and .Release.IsUpgrade $deployment)}}
```

Comment:
So if its an upgrade and the deployment with our fullName exists then just use the existing instance annotation ?

Reply:
Correct - we just carry that value over on upgrade to avoid conflicts. there is an issue getting the instance annotation value, we default to the $fullName since that is the new template value from upstream. 

Scenarios tested: 

1. **upgrade from istio 1.9.6**
    - deployed istio 1.9.6 - instance label value is rancher-istio
    - upgraded to istio 100.0.0 - instance label was carried over and is still rancher-istio
2. **upgrade from istio 1.9.6 with deleted deployment**
    - deployed istio 1.9.6 - instance label value is rancher-istio
    - deleted the kiali deployment
    - upgraded to istio 100.0.0 - instance label is now kiali
3. **upgrade from istio 100.0.0**
    - deployed istio 100.0.0 - instance label value is kiali
    - upgraded istio 100.0.0 - instance label was carried over and is still kiali
4. **upgrade from istio 100.0.0 with deleted deployment**
    - deployed istio 100.0.0 - instance label value is kiali
    - deleted the kiali deployment
    - upgraded istio 100.0.0 - instance label was carried over and is still kiali